pkg/klusterlet/clusterclaim/clusterclaimer.go

```diff
@@ -199,8 +250,8 @@ func newClusterClaim(name, value string) *clusterv1alpha1.ClusterClaim {
 	}
 }
 
-func (c *ClusterClaimer) isOpenShift() (bool, error) {
-	_, err := c.Mapper.RESTMapping(schema.GroupKind{Group: "project.openshift.io", Kind: "Project"}, "v1")
+func isOpenShift(mapper meta.RESTMapper) (bool, error) {
```

Comment:
why do we need to pass the mapper?

Reply:
That is because we need to use this function before ClusterClaimer is created.